### ISSUE TITLE
onLocationChange fires too many times

### ISSUE THREAD

**Issue by steve-cahn** [Lyokone/flutterlocation#462]
I have a listener listening to `onLocationChanged` and I set the accuracy to low and interval to 50000. The iPhone is sitting in 1 place. No matter what I do, the listener fires every second... How can I make it fire either on move or if that's not possible than to fire every x seconds?
```dart
    Location location = Location();
    location.changeSettings(accuracy: LocationAccuracy.low, interval: 50000);
    locationSubscription = location.onLocationChanged.listen((LocationData currentLocation) {
      print(currentLocation); // this runs like every second
    });
```

**Comment by Lyokone** [comment]
Hi, does it do the same thing if you `await` first the changeSettings ?
```
await location.changeSettings(accuracy: LocationAccuracy.low, interval: 50000);
```

**Comment by steve-cahn** [comment]
@Lyokone Still getting the same results

**Comment by diazsasak** [comment]
same problem. even setting distanceFilter too to 10 meters. it still fires when not moving

**Comment by anirudhsharma392** [comment]
@steve-cahn That must be happening because you are restarting the application again and again.
What exactly happens is, you have not disposed the previous instance of backgroundLocation and it's still running in the background and every time you restarts your application, another instance is created without disposing the previous one and now 2 instance is running. Because of which 2 times onLocationChanged will be called.
So basically if you have restarted the application using android studio **n** number of times, onLocationChanged  callback will be called **n** times.

**Comment by BoHellgren** [comment]
@anirudhsharma392 I don't think that is a correct explanation. I call this code only once, in initState:
```
 Location location = new Location();
  await location.changeSettings(accuracy: LocationAccuracy.high, interval: 10000, distanceFilter: 0);
  location.onLocationChanged.listen((LocationData currentLocation) {
    showPos(currentLocation.latitude, currentLocation.longitude);
  });
```
Despite interval:10000, showPos() is called once every second _when running on the web_. The underlying Geolocation.watchPosition API does not support the interval option, see https://developer.mozilla.org/en-US/docs/Web/API/PositionOptions. This must be hardcoded in the plugin, and I don't think that is done.

**Comment by danielmintz** [comment]
I have exactly the same problem

**Comment by muratmjdci** [comment]
Same issue here.. Still trying to figure it out.

**Comment by pbhuvanesh** [comment]
Has anyone been able to solve this issue? The frequency is too high I would like to get location event 10mins or so 

below is my code
`await this.location.onLocationChanged.listen((LocationData event) async {
      Entity.Location locationData = Entity.Location(
          lat: event.latitude ?? 48.8534, long: event.longitude ?? 2.3488);
      onChange(locationData);
      await location.changeSettings(
          accuracy: LocationAccuracy.powerSave,
          interval: 6000,
          distanceFilter: 3000);
    });`

**Comment by tmatrai** [comment]
I have exavtly the same problem, I get location updates at every 15 secs on IOS 14.7.1, anything I set. Do you have any suggestions? It is a mission critical one for us. 

` await locationPlugin.changeSettings(interval: 60000);`

**Comment by Jensoulus** [comment]
I am having the same issue still. Did someone solve it?

**Comment by gaiousawd** [comment]
Also have the same issue on iOS but not on android

my fix was to add a distanceFilter and the accuracy set to navigation, then add the interval, it avoid calling multiple times

**Comment by ali-eryushion** [comment]
> Also have the same issue on iOS but not on android
> 
> my fix was to add a distanceFilter and the accuracy set to navigation, then add the interval, it avoid calling multiple times

This really did work but still needs to test more to see if the issue is reverting back after some calls.
